samples/SimpleWebApp.Tests/SimpleWebAppUnitTests.cs

```diff
@@ -0,0 +1,30 @@
+using GraphQL.Conventions;
+using GraphQL.Conventions.Tests.Server;
+using GraphQL.Conventions.Tests.Server.Data.Repositories;
+using GraphQL.Conventions.Tests.Server.Schema;
+using GraphQL.Conventions.Web;
+
+namespace SimpleWebApp.Tests;
+
+public class SimpleWebAppSchemaCreationTests
+{
+    [Fact]
+    public async Task TestSimpleWebAppAsync()
+    {
+        var dependencyInjector = new DependencyInjector();
+        dependencyInjector.Register<IBookRepository>(new BookRepository());
+        dependencyInjector.Register<IAuthorRepository>(new AuthorRepository());
+
+        var requestHandler = RequestHandler
+            .New()
+            .WithDependencyInjector(dependencyInjector)
+            .WithQueryAndMutation<Query, Mutation>()
+            .Generate();
+
+        var request = Request.New(new QueryInput { QueryString = "{ __schema { types { name } } }" });
+        var result = await requestHandler.ProcessRequestAsync(request, null, dependencyInjector);
+
+        Assert.False(result.HasErrors);
```

Comment:
**FYI**: This part would previously raise errors of the sort:

>"message": "Error executing document. A different instance of the GraphType 'OutputObjectGraphType<Author>' with the name 'Author' has already been registered within the schema. Please use the same instance for all references within the schema, or use GraphQLTypeReference to reference a type instantiated elsewhere.",

and:

>"message": "Error executing document. This graph type 'OutputObjectGraphType<Book>' with name 'Book' has already been initialized. Make sure that you do not use the same instance of a graph type in multiple schemas. It may be so if you registered this graph type as singleton; see https://graphql-dotnet.github.io/docs/getting-started/dependency-injection/ for more info.",